src/components/ItemForm/ItemForm.js

```diff
@@ -153,16 +159,20 @@ class ItemForm extends React.Component {
                 {placeholder => (
                   <FormattedMessage id="ui-checkout.itemId">
                     {ariaLabel => (
-                      <Field
-                        fullWidth
-                        name="item.barcode"
-                        component={TextField}
-                        aria-label={ariaLabel}
-                        id="input-item-barcode"
-                        placeholder={placeholder}
-                        inputRef={this.barcodeEl}
-                        validationEnabled={validationEnabled}
-                      />
+                      <TitleManager prefix={(this.readyPrefix && this.state.readyToScan) ? this.readyPrefix : undefined}>
+                        <Field
+                          fullWidth
+                          name="item.barcode"
+                          component={TextField}
+                          aria-label={ariaLabel}
+                          id="input-item-barcode"
+                          placeholder={placeholder}
+                          inputRef={this.barcodeEl}
+                          validationEnabled={validationEnabled}
+                          onFocus={this.readyPrefix ? () => this.setState({ readyToScan: true }) : undefined}
+                          onBlur={this.readyPrefix ? () => this.setState({ readyToScan: false }) : undefined}
```

Comment:
An arrow function in a prop value will be recreated on every render. Instead, do something like
``` 
handleBarcodeFocus = () => this.setState({ readyToScan: true });
handleBarcodeBlur = () => this.setState({ readyToScan: false });
...
render() {
  ...
  <Field onFocus={this.readyPrefix ? this.handleBarcodeFocus : undefined} ... />
}

Reply:
Here I disagree and would advocate for readability over premature optimisation: I don't think a function definition adds significant overhead to the render whereas moving that code far from the single place it's used does hamper readability. I frequently find code in FOLIO that has me darting all over the place to see what turn out to be trivial functions like this (past-me wrote some of them, I didn't arrive at this conclusion immediately).

A more important concern with inline functions is that they don't satisfy PureComponent's equality tests and other shallow compares one might do in shouldComponentUpdate but even here I prefer readability-first and would only (maybe) focus on up-front optimisations for things like stripes-components. And you really have to measure because PureComponent can actually be slower (more comparisons) and similarly there is overhead to useCallback/useMemo.

Reply:
🤷 I come down on the other side but I don't need to start a fight over it. It sounds like you understand the risks here and have decided legibility is an OK tradeoff for now. I'm cool with that.